src/client.rs

```diff
@@ -19,8 +21,34 @@ use tokio_service::Service;
 type WireResponse<Resp, E> = Result<Result<Resp, WireError<E>>, DeserializeError>;
 type ResponseFuture<Req, Resp, E> = futures::Map<<BindClient<Req, Resp, E> as Service>::Future,
                                             fn(WireResponse<Resp, E>) -> Result<Resp, ::Error<E>>>;
-type BindClient<Req, Resp, E> =
-    <Proto<Req, Result<Resp, WireError<E>>> as ProtoBindClient<Multiplex, TcpStream>>::BindClient;
+type BindClient<Req, Resp, E> = <Proto<Req, Result<Resp, WireError<E>>> as
+                                ProtoBindClient<Multiplex, StreamType>>::BindClient;
+
+/// TLS-specific functionality
+#[cfg(feature = "tls")]
+pub mod tls {
+    use native_tls::TlsConnector;
+
+    /// TLS context
+    pub struct TlsClientContext {
```

Comment:
I don't feel super strongly about this, but since it's already in the `client::tls` module, we might consider just calling it `Context` and people can alias it if they so choose. Thoughts? Maybe the `tls` module should be top-level and it should be `tls::client::Context`?

cc @shaladdle